### PR TITLE
Add avatar round & opacity & animation feature

### PR DESCRIPTION
**Pull request by sli1989**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes have been added (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [x] Docs have been added / updated (for bug fixes / features).

## PR Type
**What kind of change does this PR introduce?**  

- [ ] Bugfix.
- [x] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number(s): N/A

## What is the new behavior?

1. refactor the avatar option, add the enable option of rounded, opacity and rotated.
3. the feature is not working in mobile.

* Screens with this changes: N/A

* Link to demo site with this changes:  http://test.saili.science/

### How to use?
In NexT `_config.yml`:
```yml
# Sidebar Avatar
avatar:
  # in theme directory(source/images): /images/avatar.gif
  # in site  directory(source/uploads): /uploads/avatar.gif
  # You can also use other linking images.
  url: #/images/avatar.gif
  # If true, the avatar would be dispalyed in circle. 
  rounded: false
  # The value of opacity should be choose from 0 to 1 to set the opacity of the avatar.
  opacity: 1
  # If true, the avatar would be rotated with the cursor.
  rotated: false
```

## Does this PR introduce a breaking change?
- [x] Yes.
- [ ] No.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -192,7 +192,12 @@ links_layout: block
 # Sidebar Avatar
 # in theme directory(source/images): /images/avatar.gif
 # in site  directory(source/uploads): /uploads/avatar.gif
-#avatar: /images/avatar.gif
+avatar:
+  rul:
```
Comment (ivan-nginx) [inline _config.yml]: Mb `url`?

File: _config.yml [hunk 2]
```diff
@@ -192,7 +192,12 @@ links_layout: block
 # Sidebar Avatar
 # in theme directory(source/images): /images/avatar.gif
 # in site  directory(source/uploads): /uploads/avatar.gif
-#avatar: /images/avatar.gif
+avatar:
+  rul: 
+  rounded: false
```
Comment (ivan-nginx) [inline _config.yml]: Some comments here?

File: _config.yml [hunk 3]
```diff
@@ -192,7 +192,12 @@ links_layout: block
...
+  # The value of opacity should be choose from 0 to 1
+  opacity: 1
+  animated: false
```
Comment (ivan-nginx) [inline _config.yml]: Some comments here too?

File: _config.yml (hunk 1 above)
Comment (sli1989) [reply _config.yml]: omg... Used to remaining useful life...

File: _config.yml (hunk 3 above)
Comment (sli1989) [reply _config.yml]: ok, any else?

Comment (ivan-nginx) [reply _config.yml]: Comments on all this parameters. As for me, IDK what mean `animated`. Is it round on 360 with cursor in or what?

Comment (sli1989) [reply _config.yml]: Just mean the rotation of the avatar with cursor.

File: _config.yml (hunk 1 above)
Comment (ivan-nginx) [reply _config.yml]: Don't understand here. It is just mistake, right?

Comment (sli1989) [reply _config.yml]: yes, it is a mistake.

File: layout/_macro/sidebar.swig [hunk 4]
```diff
@@ -29,10 +29,12 @@
       <section class="site-overview-wrap sidebar-panel{% if not display_toc or toc(page.content).length <= 1 %} sidebar-panel-active{% endif %}">
         <div class="site-overview">
           <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
-            {% if theme.avatar %}
+            {% if theme.avatar.url %}
+              <a href="/" class="site-author-image" rel="start" style="border:none">
```
Comment (ivan-nginx) [inline layout/_macro/sidebar.swig]: Ok, what about `a href`? Is it needed or mb not? And if yes, mb need to add additional option to redefine this url?

Comment (sli1989) [reply layout/_macro/sidebar.swig]: I prefer to add the function `clicking the avatar to homepage` as the inherent function. Besides, i don't know how to add this in `if condition`. 😢

Comment (sli1989) [reply layout/_macro/sidebar.swig]: You mean 

- the default to homepage?
- the custom to custom linking?

Comment (ivan-nginx) [reply layout/_macro/sidebar.swig]: No, i mean:

* the default for nothing (as it for now)
* custom - is custom; if user want, he can define link

Comment (sli1989) [reply layout/_macro/sidebar.swig]: How about this one?

- the default for nothing, but linking to the present page.
- user can custom the external url of avatar.

Comment (ivan-nginx) [reply layout/_macro/sidebar.swig]: No need to use `exturl` option.

Comment (sli1989) [reply layout/_macro/sidebar.swig]: well, I delete the linking function of avatar.

**Pull request merged**